### DISCUSSION TITLE
Plan to write a paper for FSRS

### DISCUSSION THREAD

**Discussion by L-M-Sherlock** [open-spaced-repetition/free-spaced-repetition-scheduler#16] [General]
FSRS has been released since one and half years ago. Many things happened and my understanding about memory model is also improved. I find out many incorrect practices in spaced repetition algorithm researches (including the first paper by myself). So I want to write a new paper about FSRS and other findings, such as the metrics, evaluation, simulation and so on.

The paper will help researchers learn about FSRS, provide proofs for softwares using FSRS and promote the progress in that field.

Here I'd like to call for co-authors, reviewers and supporters. Any suggestion and contribution is welcome.

For the dataset, is it permissible to use the revlogs collected from Anki users for my research? @dae

For the co-authors, @Expertium @user1823, you help me a lot in improving FSRS, so I want to add your names in the acknowledgement section. It's nicer if you want to contribute to the paper directly (and to be co-authors).

**Comment by dae** [comment]
>  is it permissible to use the revlogs collected from Anki users for my research?

Yes, but if you need to provide the dataset to others when sharing the research, please make it clear that people will need to ask permission before using the dataset for other things.

**Reply by L-M-Sherlock** [reply]
Is it possible to publish the dataset in Dataverse? For example, the dataset of MaiMemo is released on https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi:10.7910/DVN/VAGUL0. We can write License/Data Use Agreement for the dataset.

**Reply by dae** [reply]
If we can list a custom license, then it's ok. There's some ambiguity about "non-commercial" in the CC license. I'd like to make the data available for university academic work / papers, but not available for use by company-backed research/development.

**Reply by L-M-Sherlock** [reply]
It's nice if you could write the license or upload the dataset because English is not my native language and I'm not familiar with law terms.

**Reply by dae** [reply]
I'm not a lawyer, but how about this:

This data was provided to the FSRS project for use in memory research. University students and individuals may also use it for their own research. Please do not redistribute the data publicly (other than linking here). If you wish to use the data in other contexts, please request permission on https://anki.tenderapp.com.

**Reply by L-M-Sherlock** [reply]
I hosted the dataset at huggingface.co: https://huggingface.co/datasets/open-spaced-repetition/FSRS-Anki-20k

**Comment by Expertium** [comment]
I'd love to contribute to the paper if I can, of course.

**Comment by user1823** [comment]
I would also love to contribute.

**Comment by jiweiyuan** [comment]
I want to be reviewer of this work

**Comment by joshdavham** [comment]
I'm also down to review if you're looking for more help. My native language is English and I've been meaning to study the internals of FSRS for a long time.

**Comment by RickCarlino** [comment]
Not sure how I can help, but please reach out if there's anything I can do to support the project. 

 * Software developer by day
 * Implemented a typescript FSRS implementation based on the previous paper.
 * Native English speaker.
 * I use FSRS every day in a custom application I built.
 * My contact information is very easy to find :wink:

**Comment by Expertium** [comment]
So did you put off the idea of writing a paper?

**Reply by L-M-Sherlock** [reply]
Yes. I feel tired.

**Comment by L-M-Sherlock** [comment]
I apologize for the lack of progress over nearly a year. Let me explain the current situation:

Regarding the FSRS paper, I'm uncertain how to begin. Most of FSRS's improvements were based on trial and error, making it difficult to describe with a clear narrative. Furthermore, FSRS is more of an engineering product than scientific research. I've also lost interest in paper writing, as it doesn't significantly contribute to adopting FSRS. Moreover, our community's research is conducted in the public domain, allowing everyone to trace each advancement and related discussion of FSRS. Therefore, I've decided to cancel the plan for writing this paper.

**Reply by Expertium** [reply]
I think it would be good because then more people would know about FSRS. But ok.

**Reply by andymatuschak** [reply]
For what little it's worth: I admire this decision, and I want to share some encouragement!

Surrounded by an academic environment, the "default" thing to do to "get credit" for one's work is to write and publish a refereed paper. But it's often not at all the right thing to do, either for the idea or for one's own creative practice. It's hard to break out of that kind of mold.

I think you're right that paper writing will have no meaningful effect on the adoption of FSRS. It's hard to see an argument that it would. The most significant thing you could have done to get widespread adoption was to integrate it into a popular system, which is exactly what you've done—and it's about to become the default. That sounds like a great outcome to me. And that work is public, so others can benefit from what you've learned. (I have to say: I'm skeptical that transformative progress for SRS-as-we-know-it scheduling algorithms is likely to come from other academics.)

All this said: if you reframe away from "publish an academic paper", I wonder if there's any value in a frame like "writing up your biggest learnings in an informal article"? That is: perhaps the job isn't to "help researchers learn about FSRS" (it being integrated in Anki will do that!); or "provide proofs for software using FSRS" (Anki demonstrates that!); or even necessarily to "promote progress"—but to distill your much improved understanding?

**Reply by Expertium** [reply]
@andymatuschak this is totally random, but [here](https://notes.andymatuschak.org/zYK41LaWjuWkRmv3ZGnNmoF), in "Comparison of contemporary implementations by Jarrett Ye and collaborators", I recommend you to link to https://github.com/open-spaced-repetition/srs-benchmark?tab=readme-ov-file#srs-benchmark rather than to my Reddit post (which will be outdated anyway once I make a new one).
